cinema/serializers.py

```diff
@@ -78,3 +98,58 @@
     class Meta:
         model = MovieSession
         fields = ("id", "show_time", "movie", "cinema_hall")
+
+    def to_representation(self, instance):
+        representation = super().to_representation(instance)
+        representation["taken_places"] = [
+            {"row": ticket.row, "seat": ticket.seat}
+            for ticket in instance.tickets.all()
+        ]
+        return representation
+
+
+class TicketSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Ticket
+        fields = ("id", "row", "seat", "movie_session", "order")
+
+    def validate(self, attrs):
+        data = super(TicketSerializer, self).validate(attrs)
+        cinema_hall = attrs["movie_session"].cinema_hall
+        Ticket.validate_position(
+            attrs["row"],
+            cinema_hall.rows,
+            "row",
+            serializers.ValidationError
+        )
+        Ticket.validate_position(
+            attrs["seat"],
+            cinema_hall.seats_in_row,
+            "seat",
+            serializers.ValidationError
+        )
+        return data
+
+
+class OrderSerializer(serializers.ModelSerializer):
+    tickets = TicketSerializer(many=True, read_only=False, allow_empty=False)
+
+    class Meta:
+        model = Order
+        fields = ("id", "created_at", "user", "tickets")
+
+    def create(self, validated_data):
+        with transaction.atomic():
+            tickets_data = validated_data.pop("tickets")
+            order = Order.objects.create(**validated_data)
+            for ticket_data in tickets_data:
+                Ticket.objects.create(order=order, **ticket_data)
+            return order
```

Comment:
The `create` method in `OrderSerializer` uses a transaction to ensure atomicity when creating an order and its associated tickets. This is a good practice to maintain data integrity.